### ISSUE TITLE
ec in trait Security of type play.libs.concurrent.HttpExecutionContext is not defined

### ISSUE THREAD

**Issue by leleuj** [pac4j/play-pac4j#132]
@entropicdrifter It seems that since this commit: https://github.com/pac4j/play-pac4j/commit/194d85f341ec01274304367262735a690afaf6d1, the [play-pac4j-scala-demo](https://github.com/pac4j/play-pac4j-scala-demo) demo returns the following error:

```
[info] Compiling 22 Scala sources and 2 Java sources to /Users/jleleu/sources/play-pac4j-scala-demo/target/scala-2.11/classes...
[error] /Users/jleleu/sources/play-pac4j-scala-demo/app/controllers/Application.scala:20: class Application needs to be abstract, since value ec in trait Security of type play.libs.concurrent.HttpExecutionContext is not defined
[error] class Application @Inject() (val config: Config, val playSessionStore: PlaySessionStore) extends Controller with Security[CommonProfile] {
[error]       ^
[error] /Users/jleleu/sources/play-pac4j-scala-demo/app/controllers/ApplicationWithFilter.scala:18: class ApplicationWithFilter needs to be abstract, since value ec in trait Security of type play.libs.concurrent.HttpExecutionContext is not defined
[error] class ApplicationWithFilter @Inject() (val config: Config, val playSessionStore: PlaySessionStore) extends Controller with Security[CommonProfile] {
[error]       ^
[error] two errors found
[info] Compiling 22 Scala sources and 2 Java sources to /Users/jleleu/sources/play-pac4j-scala-demo/target/scala-2.11/classes...
[error] /Users/jleleu/sources/play-pac4j-scala-demo/app/controllers/Application.scala:20: class Application needs to be abstract, since value ec in trait Security of type play.libs.concurrent.HttpExecutionContext is not defined
[error] class Application @Inject() (val config: Config, val playSessionStore: PlaySessionStore) extends Controller with Security[CommonProfile] {
[error]       ^
[error] /Users/jleleu/sources/play-pac4j-scala-demo/app/controllers/ApplicationWithFilter.scala:18: class ApplicationWithFilter needs to be abstract, since value ec in trait Security of type play.libs.concurrent.HttpExecutionContext is not defined
[error] class ApplicationWithFilter @Inject() (val config: Config, val playSessionStore: PlaySessionStore) extends Controller with Security[CommonProfile] {
[error]       ^
[error] two errors found
[error] application -
[info]
[info] ! @719anl7j2 - Internal server error, for (GET) [/] ->
[info]
[info] play.forkrun.protocol.Serializers$playExceptionPickler$$anon$1: Compilation error[class Application needs to be abstract, since value ec in trait Security of type play.libs.concurrent.HttpExecutionContext is not defined]
[info] 	at play.forkrun.protocol.Serializers$playExceptionPickler$.unpickle(Serializers.scala:159)
[info] 	at play.forkrun.protocol.Serializers$PlayRunsupportReloaderCompileFailureUnpickler$macro$56$2$.unpickle(Serializers.scala:175)
[info] 	at play.forkrun.protocol.Serializers$PlayRunsupportReloaderCompileResultUnpickler$macro$66$2$.unpickle(Serializers.scala:181)
[info] 	at scala.pickling.Unpickler$class.unpickleEntry(Pickler.scala:79)
[info] 	at play.forkrun.protocol.Serializers$PlayRunsupportReloaderCompileResultUnpickler$macro$66$2$.unpickleEntry(Serializers.scala:181)
[info] 	at scala.pickling.functions$.unpickle(functions.scala:11)
[info] 	at sbt.serialization.JsonValue$$anonfun$parse$1.apply(SerializedValue.scala:104)
[info] 	at scala.util.Try$.apply(Try.scala:192)
[info] 	at sbt.serialization.JsonValue.parse(SerializedValue.scala:104)
[info] 	at sbt.protocol.BuildValue.value(Values.scala:11)
```

Can you investigate please?

**Comment by entropicdrifter** [comment]
Okay. Is this in the 2.4.x branch, or master? The commit you linked to was in master, but you labelled it 2.4.0

Either way, I will work on both.

**Comment by leleuj** [comment]
In master for sure. I didn't test it on 2.4.x, but as both are based on Play 2.5, I assume the problem occurs also.

**Comment by entropicdrifter** [comment]
I think I fixed it. I'll need to make a PR for both the play-pac4j plugin and the play-pac4j-scala-demo.
That's for the master branch, I'll get to the 2.4.x branch next.

**Comment by leleuj** [comment]
Thanks

**Comment by entropicdrifter** [comment]
I tested 2.4.x, but it seems that it already works! 

I think the difference might be because in 2.5.x the Injection scheme is different due to the PlaySessionStore.

The current play-pac4j-scala-demo 2.4.x branch pulls in the 2.4.0 release (I assume it is supposed to until the release of 2.4.1), but I deleted that line from build.sbt and pulled in the current play-pac4j 2.4.x and it still worked perfectly, from what I can tell.
